activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -384,7 +384,30 @@ def insert_fixtures(fixtures, table_name)
         manager.into(table)
         columns.each_key { |column| manager.columns << table[column] }
         manager.values = manager.create_values_list(values)
-        execute manager.to_sql, "Fixtures Insert"
+        manager.to_sql
+      end
+
+
+      # Generate and execute a single SQL string that contains multiple INSERT statement for *all* fixtures.
+      # INSERT INTO table (col1, col2) VALUES (?, ?), (?, DEFAULT);\n
+      # INSERT INTO table2 (col1, col2) VALUES (?, ?), (?, DEFAULT);\n
+      #
+      # Overridden in adapters that require something beyond a simple insert (eg. Oracle).
+      def insert_fixtures(fixture_set, tables_to_delete = [])
```

Comment:
In line with @sgrif's comment: this is public API.

Reply:
Ah right, then I guess I don't have a choice but to split this method. This would have the downside of creating another git churn :(

```ruby
def build_sql
  returns the sql string
end

def insert_fixtures
  build_sql
  execute
end

def insert_multiple_fixtures # new method to generate a single multi query statement
  total_sql = ''.dup

  fixtures.each do
    total_sql << build_sql
  end

  execute
end
```

Reply:
I think we will need to do it. Call the new method that does the multi insert `insert_fixtures_set`.